worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongePlatform.java

```diff
@@ -0,0 +1,190 @@
+        List<com.sk89q.worldedit.world.World> ret = new ArrayList<>(worlds.size());
+        for (org.spongepowered.api.world.World world : worlds) {
+            ret.add(new SpongeNMSWorld(world));
+        }
+        return ret;
+    }
+
+    @Nullable
+    @Override
+    public Player matchPlayer(Player player) {
+        if (player instanceof SpongePlayer) {
+            return player;
+        } else {
+            Optional<org.spongepowered.api.entity.living.player.Player> optPlayer = Sponge.getServer().getPlayer(player.getUniqueId());
+            return optPlayer.isPresent() ? new SpongePlayer(this, optPlayer.get()) : null;
+        }
+    }
+
+    @Nullable
+    @Override
+    public World matchWorld(World world) {
+        if (world instanceof SpongeWorld) {
+            return world;
+        } else {
+            for (org.spongepowered.api.world.World ws : Sponge.getServer().getWorlds()) {
+                if (ws.getName().equals(world.getName())) {
+                    return new SpongeNMSWorld(ws);
+                }
+            }
+
+            return null;
+        }
+    }
+
+    @Override
+    public void registerCommands(Dispatcher dispatcher) {
+        for (CommandMapping command : dispatcher.getCommands()) {
+            CommandAdapter adapter = new CommandAdapter(command) {
+                @Override
+                public CommandResult process(CommandSource source, String arguments) throws org.spongepowered.api.command.CommandException {
+                    CommandEvent weEvent = new CommandEvent(SpongeWorldEdit.inst().wrapCommandSource(source), command.getPrimaryAlias() + " " + arguments);
+                    WorldEdit.getInstance().getEventBus().post(weEvent);
+                    return weEvent.isCancelled() ? CommandResult.success() : CommandResult.empty();
+                }
+
+                @Override
+                public List<String> getSuggestions(CommandSource source, String arguments) throws org.spongepowered.api.command.CommandException {
+                    CommandSuggestionEvent weEvent = new CommandSuggestionEvent(SpongeWorldEdit.inst().wrapCommandSource(source), command.getPrimaryAlias() + " " + arguments);
+                    WorldEdit.getInstance().getEventBus().post(weEvent);
+                    return weEvent.getSuggestions();
+                }
+            };
+            Sponge.getCommandManager().register(SpongeWorldEdit.inst(), adapter, command.getAllAliases());
+        }
+    }
+
+    @Override
+    public void registerGameHooks() {
+        // We registered the events already anyway, so we just 'turn them on'
+        hookingEvents = true;
+    }
+
+    @Override
+    public SpongeConfiguration getConfiguration() {
+        return mod.getConfig();
+    }
+
+    @Override
+    public String getVersion() {
+        return mod.getInternalVersion();
+    }
+
+    @Override
+    public String getPlatformName() {
+        return "Sponge-Official";
```

Comment:
This is just convention:
https://github.com/sk89q/WorldEdit/blob/master/worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/BukkitServerInterface.java#L156

---- On Wed, 18 May 2016 17:34:37 -0700 notifications@github.com wrote ----

In worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongePlatform.java:

> +        hookingEvents = true;
> +    }
> +
> +    @Override
> +    public SpongeConfiguration getConfiguration() {
> +        return mod.getConfig();
> +    }
> +
> +    @Override
> +    public String getVersion() {
> +        return mod.getInternalVersion();
> +    }
> +
> +    @Override
> +    public String getPlatformName() {
> +        return "Sponge-Official";
...as opposed to what? Why the 'Official'?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub